File: algorithm/500. Keyboard Row/500.cpp
class Solution {
public:
    vector<string> findWords(vector<string>& words) {
        unordered_set<char> row1 {'q', 'w', 'e', 'r', 't', 'y','u', 'i', 'o', 'p'};
        unordered_set<char> row2 {'a', 's', 'd', 'f', 'g', 'h', 'j', 'k', 'l'}; 
        unordered_set<char> row3 { 'z', 'x', 'c', 'v', 'b' ,'n', 'm'};
        vector<unordered_set<char>> rows {row1, row2, row3};
        
        vector<string> result;
        for (int i=0 ;i<words.size();i++)
        {
            int row=0;
            for(int j=0;j<3;j++)
            {
                if(rows[j].count((char)tolower(words[i][0])))
                    row=j;
            }
            result.push_back(words[i]);
            for(int j=1;j<words[i].size();j++)
            {
                if(rows[row].count((char)tolower(words[i][j]))==0)
                {
                    result.pop_back();
                    break;
                }
            }
        }
        return result;
    }
};

File: algorithm/500. Keyboard Row/500.py
class Solution(object):
    def findWords(self, words):
        """
        :type words: List[str]
        :rtype: List[str]
        """
        row1=['q', 'w', 'e', 'r', 't', 'y','u', 'i', 'o', 'p'];
        row2=['a', 's', 'd', 'f', 'g', 'h', 'j', 'k', 'l']; 
        row3=[ 'z', 'x', 'c', 'v', 'b' ,'n', 'm'];
        row=[row1,row2,row3]
        result=[]
        rownum=0
        for i in range(len(words)):
            for j in range(3):
                if row[j].count((words[i][0].lower()))>0:
                    rownum=j
            result.append(words[i])
            for j in range(1,len(words[i])):
                if row[rownum].count(words[i][j].lower())==0:
                    result.pop()
                    break
        
        return result
